Makefile

```diff
@@ -2163,17 +2161,6 @@ $(BUILD_DIR)/rtti_ok:
 	@exit 1
 endif
 
-.PHONY: doc
```

Comment:
What's the story here?

Reply:
The top-level Doxyfile and Doxyfile.in were... confusing. I'm not sure `Doxyfile` was actually used for anything. The Makefile just sed-ed into Doxyfile.in, but when that was removed to use CMake's newer Doxygen module, I removed it from the Makefile, too.

Reply:
See `doc/CMakeLists.txt` for the list of non-default options that were set by Doxyfile.in.

Reply:
Does this leave us unable to build the docs using make?

Reply:
I currently use make to generate docs for releases (you know, once a year or so when I actually do one). I suppose I could switch to scraping them from a CI distro, or just use cmake. But this makes make/cmake not parallel build systems. The cmake build would be necessary because we cut a feature from the make build. Not sure how I feel about that.

Reply:
Being a build target integrates better with CPack for packaging the documentation along with the binary releases.

Reply:
Building the docs is kinda niche, but as a general rule I think we should not be cutting features from the Makefile in this PR.

Reply:
@abadams - just pushed doc target to Makefile.